test/util.js

```diff
@@ -1,55 +1,86 @@
 'use strict';
 
-const fsMock = require('fs');
 const path = require('path');
+const del = require('del');
+const makeDir = require('make-dir');
+const parentModule = require('parent-module');
+const os = require('os');
+const fs = require.requireActual('fs');
 
-exports.absolutePath = str => path.join(__dirname, str);
+class TempDir {
+  dir: string;
 
-exports.mockStatIsDirectory = function mockStatIsDirectory(result) {
-  const stats = {
-    isDirectory: () => result,
-  };
+  constructor() {
+    const tempDir = fs.realpathSync(os.tmpdir());
+    const parent = parentModule();
+    const relativeParent = path.relative(process.cwd(), parent);
 
-  jest.spyOn(fsMock, 'stat').mockImplementation((path, callback) => {
-    callback(null, stats);
-  });
+    /**
+     * Each temp directory will be unique to the file
+     */
+    this.dir = path.resolve(tempDir, 'cosmiconfig', `${relativeParent}-dir`);
 
-  jest.spyOn(fsMock, 'statSync').mockImplementation(() => stats);
-};
+    // create directory
+    makeDir.sync(this.dir);
 
-exports.assertSearchSequence = function assertSearchSequence(
-  readFileMock,
-  searchPaths,
-  startCount
-) {
-  startCount = startCount || 0;
+    (this: any).absolutePath = this.absolutePath.bind(this);
+    (this: any).createDir = this.createDir.bind(this);
+    (this: any).createFile = this.createFile.bind(this);
+    (this: any).clean = this.clean.bind(this);
+    (this: any).deleteTempDir = this.deleteTempDir.bind(this);
+  }
 
-  expect(readFileMock).toHaveBeenCalledTimes(searchPaths.length + startCount);
+  absolutePath(dir: string) {
+    const absolutePath = path.join(this.dir, dir);
 
-  searchPaths.forEach((searchPath, idx) => {
-    expect(readFileMock.mock.calls[idx + startCount][0]).toBe(
-      path.join(__dirname, searchPath)
-    );
-  });
-};
+    return absolutePath;
+  }
 
-function makeReadFileMockImpl(readFile) {
-  return (searchPath, encoding, callback) => {
-    try {
-      callback(null, readFile(searchPath));
-    } catch (err) {
-      callback(err);
-    }
-  };
-}
-exports.makeReadFileMockImpl = makeReadFileMockImpl;
-
-exports.mockReadFile = function mockReadFile(sync, readFile) {
-  if (sync === true) {
-    return jest.spyOn(fsMock, 'readFileSync').mockImplementation(readFile);
-  } else {
-    return jest
-      .spyOn(fsMock, 'readFile')
-      .mockImplementation(makeReadFileMockImpl(readFile));
+  createDir(dir: string) {
+    const dirname = this.absolutePath(dir);
+    makeDir.sync(dirname);
+  }
+
+  createFile(file: string, contents: string | Object) {
```

Comment:
Contents is always a string right? `${contents}` would not work well either if it was an object.

Reply:
Yes, good catch! `Object` was added from a previous implementation that I forgot to remove.